### PR TITLE
If the ssh agent hasn't been given, then check for qrsh and friends

### PR DESCRIPTION
**Pull request by rhc54**
(cherry picked from commit open-mpi/ompi@1c44543854bad60448bfa68c6ae79e830cb8df7e)

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/2085/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/2087/ for details.

**Comment by jsquyres** [comment]
@rhc54 I note that this is quite a bit more than the cherry-picked https://github.com/open-mpi/ompi/commit/1c44543854bad60448bfa68c6ae79e830cb8df7e.  Was that intentional?

**Comment by jsquyres** [comment]
Doing a little log spelunking, it looks like this also includes open-mpi/ompi@5717b75b453365cc67b148b5cd2c3c25f8eafe43.

File: orte/mca/plm/rsh/plm_rsh_component.c [hunk 1]
```diff
@@ -258,8 +258,10 @@ static int rsh_component_query(mca_base_module_t **module, int *priority)
     char *tmp;
 
     /* Check if we are under Grid Engine parallel environment by looking at several
-     * environment variables.  If so, setup the path and argv[0]. */
-    if (NULL == mca_plm_rsh_component.agent) {
+     * environment variables.  If so, setup the path and argv[0].
+     * Note that we allow the user to specify the launch agent
+     * even if they are in a Grid Engine environment */
+    if (0 == strcmp(mca_plm_rsh_component.agent, "ssh : rsh")) {
```
Comment (jsquyres) [inline orte/mca/plm/rsh/plm_rsh_component.c:264]: Minor nit: https://github.com/open-mpi/ompi/pull/1972 might be a more robust way to check to see if we're not using the default value...?

File: orte/mca/plm/rsh/plm_rsh_module.c [hunk 2]
```diff
@@ -177,14 +178,59 @@ static void process_launch_list(int fd, short args, void *cbdata);
...
  */
 static int rsh_init(void)
 {
+    char *tmp;
     int rc;
 
+    /* we were selected, so setup the launch agent */
+    if (mca_plm_rsh_component.using_qrsh) {
+        /* perform base setup for qrsh */
+        asprintf(&tmp, "%s/bin/%s", getenv("SGE_ROOT"), getenv("ARC"));
```
Comment (jsquyres) [inline orte/mca/plm/rsh/plm_rsh_module.c]: Should there be NULL checks for these `getenv()`s to ensure we don't crash and burn?

**Comment by jsquyres** [comment]
Other than the two minor nits (which can be fixed after this PR), as long as open-mpi/ompi@5717b75 was intended to be in this PR, 👍

File: orte/mca/plm/rsh/plm_rsh_module.c (hunk 2 above)
Comment (jsquyres) [inline orte/mca/plm/rsh/plm_rsh_module.c]: Disregard -- I see there's NULL checks for these `getenv()`s in `_component.c`; i.e., `.using_qrsh` won't be true unless they're non-NULL.

**Comment by rhc54** [comment]
I didn't worry about the history of the changes - I just fixed the problem 😄 

This should now be complete. We can add the better way of detecting default later.

**Comment by jsquyres** [comment]
You really like squashing your commits, don't you?  😄  It took a minute to figure out what was new in your new commit (i.e., the getpwent stuff).  Looks good.

I agree that the (potentially) more robust check can be added later.

**Comment by jsquyres** [comment]
@hppritcha Once CI finishes, this is good to go.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/2091/ for details.

**Comment by jsquyres** [comment]
False Travis failure (we killed the build because it was taking hours and it hadn't even start yet).

**Pull request merged**